File: 0001. Two Sum/solution.py
#!python3

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}            
        for i in range(len(nums)):
            #print(nums[i])
            complement = target - nums[i]
            if complement in dict:
                return [dict[complement], i]
            else:
                dict[nums[i]] = i
                #print(dict)
        print("No solution found!")
            

File: 0001. Two Sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> table = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            //if (table.get(complement) != null) {
            if (table.containsKey(complement)) {
                return new int[] {i, table.get(complement)};
            } else {
                table.put(nums[i], i);
            }
        }
        //return new int[] {};
        throw new IllegalArgumentException("No two sum solution");
    }
}
